exporters/trace/gcp_exporter/internal/recordable_test.cc

```diff
@@ -0,0 +1,115 @@
+#include "exporters/trace/gcp_exporter/recordable.h"
+#include <gtest/gtest.h>
+
+OPENTELEMETRY_BEGIN_NAMESPACE
+namespace exporter
+{
+namespace gcp
+{
+
+TEST(Recordable, TestSetAttribute)
+{
+    Recordable rec;
+
+    // Set 'bool' type
+    const nostd::string_view bool_key = "bool_key";
+    const common::AttributeValue bool_value = true;
+    rec.SetAttribute(bool_key, std::move(bool_value));
+
+    // Set 'integer' type
+    const nostd::string_view int_key = "int_key";
+    const int64_t seven = 7;
+    const common::AttributeValue int_value = seven;
+    rec.SetAttribute(int_key, std::move(int_value));
+
+    // Set 'string' type
+    const nostd::string_view string_key = "string_key";
+    const common::AttributeValue string_value = "test";
+    rec.SetAttribute(string_key, std::move(string_value));
+
+    auto attr_map = rec.span().attributes().attribute_map();
+    
+    EXPECT_TRUE(attr_map["bool_key"].bool_value());
+    EXPECT_EQ(seven, attr_map["int_key"].int_value());
+    EXPECT_EQ("test", attr_map["string_key"].string_value().value());
+}
+
+TEST(Recordable, TestSetIds)
+{
+    setenv("GOOGLE_CLOUD_PROJECT_ID", "test_project", 1);
+
+    const opentelemetry::trace::TraceId trace_id(
+    std::array<const uint8_t, opentelemetry::trace::TraceId::kSize>(
+    {0, 1, 0, 2, 1, 3, 1, 4, 1, 5, 1, 6, 3, 7, 0, 0}));
+
+    const opentelemetry::trace::SpanId span_id(
+    std::array<const uint8_t, opentelemetry::trace::SpanId::kSize>(
+    {1, 2, 3, 4, 5, 6, 7, 8}));
+
+    const opentelemetry::trace::SpanId parent_span_id(
+    std::array<const uint8_t, opentelemetry::trace::SpanId::kSize>(
+    {4, 5, 0, 1, 1, 1, 1, 3}));
+
+    Recordable rec;
+
+    rec.SetIds(trace_id, span_id, parent_span_id);
+
+    EXPECT_EQ("projects/test_project/traces/0001000201030104/spans/01020304", rec.span().name());
+    EXPECT_EQ("01020304", rec.span().span_id());
+    EXPECT_EQ("04050001", rec.span().parent_span_id());
+}
+
+
+TEST(Recordable, TestSetName)
+{
+    Recordable rec;
+    const nostd::string_view expected_name = "Test Span";
+    rec.SetName(expected_name);
+    EXPECT_EQ(expected_name, rec.span().display_name().value());
+}
+
+
+TEST(Recordable, TestSetStartTime)
+{
+    Recordable rec;
+
+    const std::chrono::system_clock::time_point start_time = std::chrono::system_clock::now();
+    const core::SystemTimestamp start_timestamp(start_time);
+
+    const int64_t expected_unix_start_time = std::chrono::duration_cast<std::chrono::nanoseconds>(start_time.time_since_epoch()).count();
+
+    rec.SetStartTime(start_timestamp);
+
+    const std::chrono::nanoseconds start_time_nanos(reinterpret_cast<int32_t>(rec.span().start_time().nanos()));
+    const std::chrono::seconds start_time_seconds(reinterpret_cast<int64_t>(rec.span().start_time().seconds()));
+    const std::chrono::nanoseconds unix_start_time(std::chrono::duration_cast<std::chrono::nanoseconds>(start_time_seconds).count() 
```

Comment:
Nit: I'm not sure if OpenTel has a specification on line length, if there is not, I would suggest 80 chars per line.

Reply:
Thanks! I'll try adjusting to 80 chars wherever I can.
But in some places, I believe this adjustment might affect the readability.